Testing testbed on Debian 12

lindenb1 opened the pull request:
### Preparing and Testing Testbed for Debian Bookworm Compatibility

- Related issue: https://github.com/osism/issues/issues/1028 (provides context for this work)
- Dependency: https://github.com/osism/terraform-base/pull/56 (required changes for Debian Bookworm support)

This task involves updating and testing the testbed to ensure compatibility with Debian Bookworm. We'll be verifying the deployment of various services and components.


### Deployment Status

- [x] Manager

#### Services
---
- [x] Helper Services

---
- [X] Kubernetes

---
- [x] K8s-Clusterapi

---
- [x] Ceph Services

---
- [ ] Rook Services
???

---
- [x] Infrastructure Services

---
- [x] OpenStack Services

---
- [x] Monitoring Services

---

sbstnnmnn commented:
needs: https://github.com/osism/terraform-base/pull/56

lindenb1 commented:
Netbox - Manage Ankh-Morpork location:

```

++ /usr/bin/docker inspect -f '{{.State.Health.Status}}' netbox-netbox-1
+ [[ healthy == \h\e\a\l\t\h\y ]]
+ osism netbox import
2024-05-29 09:32:17 | INFO     | Task 157a9974-9b47-4469-9b4a-95b5a544807a is running. Wait. No more output.
+ osism netbox init
2024-05-29 09:32:21 | INFO     | Task ccb3a63d-a9db-4937-bfcb-c19c77c3bc55 was prepared for execution.
2024-05-29 09:32:21 | INFO     | It takes a moment until task ccb3a63d-a9db-4937-bfcb-c19c77c3bc55 has been started and output is visible here.

PLAY [Wait for netbox service] *************************************************

TASK [Wait for netbox service] *************************************************
ok: [localhost]

PLAY [Manage sites and locations] **********************************************

TASK [Manage Discworld site] ***************************************************
changed: [localhost]

TASK [Manage Ankh-Morpork location] ********************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "\n<!DOCTYPE html>\n<html lang=\"en\">\n\n<head>\n    <title>Server Error</title>\n    <link rel=\"stylesheet\" href=\"/static/netbox-light.css\" />\n    <meta charset=\"UTF-8\">\n</head>\n\n<body>\n    <div class=\"container-fluid\">\n        <div class=\"row\">\n            <div class=\"col col-md-6 offset-md-3\">\n                <div class=\"card border-danger mt-5\">\n                    <h5 class=\"card-header\">\n                        <i class=\"mdi mdi-alert\"></i> Server Error\n                    </h5>\n                    <div class=\"card-body\">\n                        \n                            <p>\n                                There was a problem with your request. Please contact an administrator.\n                            </p>\n                        \n                        <hr />\n                        <p>\n                            The complete exception is provided below:\n                        </p>\n<pre class=\"block\"><strong>&lt;class &#x27;dcim.models.sites.Site.MultipleObjectsReturned&#x27;&gt;</strong><br />\nget() returned more than one Site -- it returned 2!\n\nPython version: 3.10.6\nNetBox version: 3.4.8</pre>\n                        <p>\n                            If further assistance is required, please post to the <a href=\"https://github.com/netbox-community/netbox/discussions\">NetBox discussion forum</a> on GitHub.\n                        </p>\n                        <div class=\"text-end\">\n                            <a href=\"/\" class=\"btn btn-primary\">Home Page</a>\n                        </div>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </div>\n</body>\n\n</html>\n"}

```


Osism tool can't pull images:

```
TASK [service-images-pull : barbican | Pull images] ****************************
Wednesday 29 May 2024  09:50:39 +0000 (0:00:02.499)       0:00:13.337 ********* 
fatal: [testbed-node-0.testbed.osism.xyz]: FAILED! => 
  msg: '[''{{ node_config_directory }}/barbican-api/:{{ container_config_directory }}/:ro'', ''/etc/localtime:/etc/localtime:ro'', "{{ ''/etc/timezone:/etc/timezone:ro'' if ansible_facts.os_family == ''Debian'' else '''' }}", ''barbican:/var/lib/barbican/'', ''kolla_logs:/var/log/kolla/'', "{{ kolla_dev_repos_directory ~ ''/barbican/barbican:/var/lib/kolla/venv/lib/python'' ~ distro_python_version ~ ''/site-packages/barbican'' if barbican_dev_mode | bool else '''' }}"]: ''dict object'' has no attribute ''os_family''. ''dict object'' has no attribute ''os_family''. [''{{ node_config_directory }}/barbican-api/:{{ container_config_directory }}/:ro'', ''/etc/localtime:/etc/localtime:ro'', "{{ ''/etc/timezone:/etc/timezone:ro'' if ansible_facts.os_family == ''Debian'' else '''' }}", ''barbican:/var/lib/barbican/'', ''kolla_logs:/var/log/kolla/'', "{{ kolla_dev_repos_directory ~ ''/barbican/barbican:/var/lib/kolla/venv/lib/python'' ~ distro_python_version ~ ''/site-packages/barbican'' if barbican_dev_mode
    | bool else '''' }}"]: ''dict object'' has no attribute ''os_family''. ''dict object'' has no attribute ''os_family''. [''{{ node_config_directory }}/barbican-api/:{{ container_config_directory }}/:ro'', ''/etc/localtime:/etc/localtime:ro'', "{{ ''/etc/timezone:/etc/timezone:ro'' if ansible_facts.os_family == ''Debian'' else '''' }}", ''barbican:/var/lib/barbican/'', ''kolla_logs:/var/log/kolla/'', "{{ kolla_dev_repos_directory ~ ''/barbican/barbican:/var/lib/kolla/venv/lib/python'' ~ distro_python_version ~ ''/site-packages/barbican'' if barbican_dev_mode | bool else '''' }}"]: ''dict object'' has no attribute ''os_family''. ''dict object'' has no attribute ''os_family''. [''{{ node_config_directory }}/barbican-api/:{{ container_config_directory }}/:ro'', ''/etc/localtime:/etc/localtime:ro'', "{{ ''/etc/timezone:/etc/timezone:ro'' if ansible_facts.os_family == ''Debian'' else '''' }}", ''barbican:/var/lib/barbican/'', ''kolla_logs:/var/log/kolla/'', "{{ kolla_dev_repos_directory ~ ''/barbican/barbican:/var/lib/kolla/venv/lib/python''
    ~ distro_python_version ~ ''/site-packages/barbican'' if barbican_dev_mode | bool else '''' }}"]: ''dict object'' has no attribute ''os_family''. ''dict object'' has no attribute ''os_family'''


PLAY RECAP *********************************************************************
2024-05-29 09:50:43 | INFO     | Play has been completed. There may now be a delay until all logs have been written.
2024-05-29 09:50:43 | INFO     | Please wait and do not abort execution.
testbed-node-0.testbed.osism.xyz : ok=3    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
testbed-node-1.testbed.osism.xyz : ok=3    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
testbed-node-2.testbed.osism.xyz : ok=3    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

Wednesday 29 May 2024  09:50:43 +0000 (0:00:04.277)       0:00:17.615 ********* 
=============================================================================== 
service-images-pull : barbican | Pull images ---------------------------- 4.28s
Group hosts based on enabled services ----------------------------------- 3.68s
Group hosts based on Kolla action --------------------------------------- 3.53s
barbican : include_tasks ------------------------------------------------ 2.50s

```

lindenb1 commented:
fatal: [testbed-node-0.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64"], "stdout": "", "stdout_lines": []}

lindenb1 commented:
Manager:
```
++ /usr/bin/docker inspect -f '{{.State.Health.Status}}' ceph-ansible
template parsing error: template: :1:8: executing "" at <.State.Health.Status>: map has no entry for key "Health"
+ [[ '' == \h\e\a\l\t\h\y ]]
+ ((  attempt_num++ == max_attempts  ))
+ sleep 5
```
Deploy:
```
++ /usr/bin/docker inspect -f '{{.State.Health.Status}}' ceph-ansible
+ [[ healthy == \h\e\a\l\t\h\y ]]
+ wait_for_container_healthy 60 kolla-ansible
+ local max_attempts=60
+ local name=kolla-ansible
+ local attempt_num=1
```


Aber bekomme immer noch den gleichen Fehler:
```
fatal: [testbed-node-2.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64"], "stdout": "", "stdout_lines": []}
fatal: [testbed-node-1.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64"], "stdout": "", "stdout_lines": []}
fatal: [testbed-node-0.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64"], "stdout": "", "stdout_lines": []}
fatal: [testbed-manager.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-21-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-21-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-21-cloud-amd64"], "stdout": "", "stdout_lines": []}
```

sbstnnmnn commented on environments/configuration.yml:
Put Debian specific variables to `environments/configuration-Debian.yml` and add appropriate line to `ansible/manager-part-3.yml` (cf. https://github.com/osism/testbed/pull/2229/commits/52307a43821a2432fd5e7a26160dc5731b0e70fa)

lindenb1 commented:
Still getting issues at the end of the deploy step ... 

```
STILL ALIVE [task 'Wait until service is available' is running] ****************
FAILED - RETRYING: [localhost]: Wait until service is available (5 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (4 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (3 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (2 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 30, "changed": false, "elapsed": 0, "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [Errno 111] Connection refused>", "redirected": false, "status": -1, "url": "https://keycloak.testbed.osism.xyz/auth/"}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
2024-06-12 11:15:04 | INFO     | Play has been completed. There may now be a delay until all logs have been written.
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
2024-06-12 11:15:04 | INFO     | Please wait and do not abort execution.

Wednesday 12 June 2024  11:15:04 +0000 (0:03:00.432)       0:03:03.504 ******** 
=============================================================================== 
Wait until service is available --------------------------------------- 180.43s
2024-06-12 11:15:04 | INFO     | Task 9b6b792f-e06d-4085-a438-09e1020a3de2 (keycloak-oidc-client-config) was prepared for execution.
2024-06-12 11:15:04 | INFO     | It takes a moment until task 9b6b792f-e06d-4085-a438-09e1020a3de2 (keycloak-oidc-client-config) has been started and output is visible here.

PLAY [Configure OIDC client for Keystone] **************************************

TASK [Wait until service is available] *****************************************
Wednesday 12 June 2024  11:15:10 +0000 (0:00:02.695)       0:00:02.695 ******** 
FAILED - RETRYING: [localhost]: Wait until service is available (30 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (29 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (28 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (27 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (26 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (25 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (24 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (23 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (22 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (21 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (20 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (19 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (18 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (17 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (16 retries left).
FAILED - RETRYING: [localhost]: Wait until service is available (15 retries left).
ok: [localhost]

TASK [Log in to Keycloak] ******************************************************
Wednesday 12 June 2024  11:16:47 +0000 (0:01:36.705)       0:01:39.401 ******** 
ok: [localhost]

TASK [Get available realms] ****************************************************
Wednesday 12 June 2024  11:16:56 +0000 (0:00:08.584)       0:01:47.985 ******** 
ok: [localhost]

TASK [Filter available realms] *************************************************
Wednesday 12 June 2024  11:17:00 +0000 (0:00:04.717)       0:01:52.702 ******** 
ok: [localhost]

TASK [Create target realm if it doesn't exist] *********************************
Wednesday 12 June 2024  11:17:02 +0000 (0:00:01.599)       0:01:54.303 ******** 
changed: [localhost]

TASK [Get available clients in realm] ******************************************
Wednesday 12 June 2024  11:17:08 +0000 (0:00:05.899)       0:02:00.202 ******** 
ok: [localhost]

TASK [Filter available clients in realm] ***************************************
Wednesday 12 June 2024  11:17:11 +0000 (0:00:03.727)       0:02:03.929 ******** 
ok: [localhost]

TASK [Create OIDC client configuration] ****************************************
Wednesday 12 June 2024  11:17:13 +0000 (0:00:01.724)       0:02:05.653 ******** 
changed: [localhost]

TASK [Get internal ID for client keystone] *************************************
Wednesday 12 June 2024  11:17:18 +0000 (0:00:04.691)       0:02:10.345 ******** 
ok: [localhost]

TASK [Filter internal ID for client keystone] **********************************
Wednesday 12 June 2024  11:17:22 +0000 (0:00:03.786)       0:02:14.132 ******** 
ok: [localhost]

TASK [Get available mappers for client] ****************************************
Wednesday 12 June 2024  11:17:23 +0000 (0:00:01.142)       0:02:15.275 ******** 
ok: [localhost]

TASK [Filter available mappers for client] *************************************
Wednesday 12 June 2024  11:17:27 +0000 (0:00:03.936)       0:02:19.211 ******** 
ok: [localhost]

TASK [Create mappers for client] ***********************************************
Wednesday 12 June 2024  11:17:28 +0000 (0:00:01.378)       0:02:20.590 ******** 
changed: [localhost] => (item=openstack-user-domain)
changed: [localhost] => (item=openstack-default-project)

TASK [Get available components in realm] ***************************************
Wednesday 12 June 2024  11:17:35 +0000 (0:00:06.404)       0:02:26.995 ******** 
ok: [localhost]

TASK [Filter available components in realm] ************************************
Wednesday 12 June 2024  11:17:38 +0000 (0:00:03.627)       0:02:30.623 ******** 
ok: [localhost]

TASK [Add privateKey and certificate to realm] *********************************
Wednesday 12 June 2024  11:17:39 +0000 (0:00:01.311)       0:02:31.935 ******** 
fatal: [localhost]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
localhost                  : ok=15   changed=3    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

2024-06-12 11:17:42 | INFO     | Play has been completed. There may now be a delay until all logs have been written.
2024-06-12 11:17:42 | INFO     | Please wait and do not abort execution.
Wednesday 12 June 2024  11:17:42 +0000 (0:00:02.407)       0:02:34.342 ******** 
=============================================================================== 
Wait until service is available ---------------------------------------- 96.71s
Log in to Keycloak ------------------------------------------------------ 8.58s
Create mappers for client ----------------------------------------------- 6.40s
Create target realm if it doesn't exist --------------------------------- 5.90s
Get available realms ---------------------------------------------------- 4.72s
Create OIDC client configuration ---------------------------------------- 4.69s
Get available mappers for client ---------------------------------------- 3.94s
Get internal ID for client keystone ------------------------------------- 3.79s
Get available clients in realm ------------------------------------------ 3.73s
Get available components in realm --------------------------------------- 3.63s
Add privateKey and certificate to realm --------------------------------- 2.41s
Filter available clients in realm --------------------------------------- 1.72s
Filter available realms ------------------------------------------------- 1.60s
Filter available mappers for client ------------------------------------- 1.38s
Filter available components in realm ------------------------------------ 1.31s
Filter internal ID for client keystone ---------------------------------- 1.14s
make[1]: *** [Makefile:115: deploy] Error 2
make[1]: Leaving directory '/home/claris/osism/osism/nobel-testbed/testbed/terraform'
make: *** [Makefile:108: deploy] Error 2

```

sbstnnmnn commented:
```
++ /usr/bin/docker inspect -f '{{.State.Health.Status}}' ceph-ansible
template parsing error: template: :1:8: executing "" at <.State.Health.Status>: map has no entry for key "Health"
+ [[ '' == \h\e\a\l\t\h\y ]]
+ ((  attempt_num++ == max_attempts  ))
+ sleep 5
++ /usr/bin/docker inspect -f '{{.State.Health.Status}}' ceph-ansible
template parsing error: template: :1:8: executing "" at <.State.Health.Status>: map has no entry for key "Health"
+ [[ '' == \h\e\a\l\t\h\y ]]
+ ((  attempt_num++ == max_attempts  ))
+ return 1
make[1]: *** [Makefile:125: deploy-manager] Error 1
```

sbstnnmnn commented:
> drivetemp is currently not activatable at the testbed, despite that integration-test's are working fine:
> 
> ```
> fatal: [testbed-node-0.testbed.osism.xyz]: FAILED! => {"changed": false, "msg": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "name": "drivetemp", "params": "", "rc": 1, "state": "present", "stderr": "modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64\n", "stderr_lines": ["modprobe: FATAL: Module drivetemp not found in directory /lib/modules/6.1.0-18-cloud-amd64"], "stdout": "", "stdout_lines": []}
> ```

Got this error again.

sbstnnmnn commented:
```
TASK [osism.services.nexus : Deleting script create_repos_from_list] ***********
Monday 08 July 2024  14:08:40 +0000 (0:00:02.768)       0:00:41.744 *********** 
fatal: [testbed-manager.testbed.osism.xyz]: FAILED! => {"changed": false, "connection": "close", "content_length": "0", "date": "Mon, 08 Jul 2024 14:08:41 GMT", "elapsed": 0, "msg": "Status code was 401 and not [204, 404]: HTTP Error 401: Unauthorized", "redirected": false, "server": "Nexus/3.69.0-02 (OSS)", "status": 401, "url": "https://nexus.testbed.osism.xyz/service/rest/v1/script/create_repos_from_list", "www_authenticate": "BASIC realm=\"Sonatype Nexus Repository Manager\"", "x_content_type_options": "nosniff"}
```

sbstnnmnn commented:
Docker community galaxy packages are offline:

https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/community-docker-3.10.4.tar.gz

File does not exist anymore on AWS.

lindenb1 commented:
Debian 12 is now getting deployed with all major stages:


- 001-helper-services.sh
- 005-kubernetes.sh
- 006-kubernetes-clusterapi.sh
- 100-ceph-services.sh
- 200-infrastructure-services.sh
- 300-openstack-services.sh
- 400-monitoring-services.sh
